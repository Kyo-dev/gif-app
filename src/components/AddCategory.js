import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length >= 3) {
      setCategories((cat) => [inputValue, ...cat]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        value={inputValue}
        type="text"
        onChange={handleInputChange}
        placeholder="Search..."
      />
    </form>
  );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}