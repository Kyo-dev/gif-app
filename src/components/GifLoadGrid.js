import React from "react";
import PropTypes from 'prop-types'

export const GifLoadGrid = ({ title, gif }) => {
  return (
    <div className="card animate__animated animate__backInUp">
      console.log(gif)
      <img src={gif} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifLoadGrid.propTypes = {
  title: PropTypes.string.isRequired,
  gif: PropTypes.string.isRequired
}