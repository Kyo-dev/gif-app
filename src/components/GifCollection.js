import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifLoadGrid } from "./GifLoadGrid";
import PropTypes from 'prop-types';

export const GifCollection = ({ category }) => {
  const {data:gifs, loading} = useFetchGifs(category);
  return (
    <>
      <h3 className= "card animate__animated animate__backInDown">{category}</h3>
      {loading && <p className ="card animate__animated animate__flash">Loading...</p> }
      <div className="card-container">
        {gifs.map((img) => (
          <GifLoadGrid key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifCollection.propTypes = {
  category: PropTypes.string.isRequired
}