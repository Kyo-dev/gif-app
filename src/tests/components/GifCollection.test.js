import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { GifCollection } from "../../components/GifCollection";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");
const category = "samurai X";

describe("test in <GrifCollection />", () => {
  test("Show component OK", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifCollection category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("load images", () => {
    const gifs = [
      {
        id: "ABC",
        title: "This is a test",
        gif: "https://localhost/testing.jpg",
      },
      {
        id: "123",
        title: "This is a test",
        gif: "https://localhost/testing.jpg",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifCollection category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").at(0).exists()).toBe(false);
    expect(wrapper.find("GifLoadGrid").length).toBe(gifs.length);
  });
});
