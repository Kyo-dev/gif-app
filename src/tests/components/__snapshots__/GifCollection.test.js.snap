// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GrifCollection /> Show component OK 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__backInDown"
  >
    samurai X
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-container"
  />
</Fragment>
`;

exports[`test in <GrifCollection /> load images 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__backInDown"
  >
    samurai X
  </h3>
  <div
    className="card-container"
  >
    <GifLoadGrid
      gif="https://localhost/testing.jpg"
      id="ABC"
      key="ABC"
      title="This is a test"
    />
    <GifLoadGrid
      gif="https://localhost/testing.jpg"
      id="123"
      key="123"
      title="This is a test"
    />
  </div>
</Fragment>
`;
